### ISSUE TITLE
objectid sometimes incremented

### ISSUE THREAD

**Issue by n-0** [mongodb/bson-rust#380]
## Versions/Environment
1. What version of Rust are you using? 
rustc 1.61.0
2. What operating system are you using?
Fedora 36
3. What versions of the driver and its dependencies are you using? 
mongodb:2.2.2
bson:2.4.0
4. What version of MongoDB are you using? 
5.0.9 with the official docker image 
5. What is your MongoDB topology
standalone

## Describe the bug
I have a collection (let's call it collection_info) with a property `collection_name` that has the form
`collection-{objectid-hex-representation}` e.g.
`collection-63530e1c4d372a8795e46fcd`. It stores the name for another collection.
When creating a document in collection_info it can happen that this becomes `collection-63530e1c4d372a8795e46fce`.
The name is generated via
```rust
let collection_id = ObjectId::new();
let collection_name = format!("collection-{:#?}", collection_id.to_string());
```
In the same way when querying the property collection_name I get
`collection-63530e1c4d372a8795e46fcd`, but when I insert into the other collection via 
`db.collection::<OtherCollectionDocumentType>(&collection_name).insert_one(a_valid_document), None).await`
the actual collection_name used becomes `collection-63530e1c4d372a8795e46fce`
The spooky part is, that this behavior stops as soon as I log what is queried
`log::info!("Used collection with name {:#?}", collection_name);`
I don't want the ObjectId's hex to be incremented, and because this happens only sometimes
and can be fixed by logging, it is tricky to debug. 
My question is also if this event can be somehow triggered by using
functions like `clone` or `to_string` on ObjectIds?

**Comment by isabelatkinson** [comment]
Hi @n-0, thanks for filing this issue! Both `clone` and `to_string` take `&self` as a parameter (i.e. an immutable reference), so it would not be possible for our implementations of these methods to mutate the `ObjectId`. Do you have a more detailed reproduction of this bug that you can share?

**Comment by n-0** [comment]
I think I've found the issue and it doesn't originate from `bson-rust` but rather from me missing out on `ObjectId::new()` being called twice, (once for the `collection_name` and another time for `collection_id`) in the code on which the given example is based. I removed the second call without noticing and introduced more logging in the same commit, misleading me to think that the logging fixed it. I close this issue. Thank you for still taking the time ^^
